--- a/hw/xfree86/common/xf86VGAarbiter_priv.h
+++ b/hw/xfree86/common/xf86VGAarbiter_priv.h
@@ -23,7 +23,7 @@
 static inline void xf86VGAarbiterInit() {}
 static inline void xf86VGAarbiterFini() {}
 static inline void xf86VGAarbiterScrnInit(ScrnInfoPtr pScrn) {}
-static inline void xf86VGAarbiterWrapFunctions(void) { return FALSE; }
+static inline Bool xf86VGAarbiterWrapFunctions(void) { return FALSE; }
 static inline void xf86VGAarbiterLock(ScrnInfoPtr pScrn) {}
 static inline void xf86VGAarbiterUnlock(ScrnInfoPtr pScrn) {}
 
