diff -urN xserver-xlibre-xserver-25.0.0.22.orig/hw/vfb/InitOutput.c xserver-xlibre-xserver-25.0.0.22/hw/vfb/InitOutput.c
--- xserver-xlibre-xserver-25.0.0.22.orig/hw/vfb/InitOutput.c	2026-04-21 20:22:22.000000000 +0200
+++ xserver-xlibre-xserver-25.0.0.22/hw/vfb/InitOutput.c	2026-05-20 07:57:01.033177374 +0200
@@ -75,6 +75,15 @@
 #include "glx_extinit.h"
 #include "randrstr.h"
 
+#ifdef VFB_NO_GLX
+void
+GlxExtensionInit(void)
+{
+}
+
+Bool noGlxExtension = TRUE;
+#endif
+
 #define VFB_DEFAULT_WIDTH      1280
 #define VFB_DEFAULT_HEIGHT     1024
 #define VFB_DEFAULT_DEPTH        24
@@ -1088,7 +1097,9 @@
         vfbPixmapDepths[32] = TRUE;
     }
 
+#ifndef VFB_NO_GLX
     xorgGlxCreateVendor();
+#endif
 
     for (i = 1; i <= 32; i++) {
         if (vfbPixmapDepths[i]) {
diff -urN xserver-xlibre-xserver-25.0.0.22.orig/hw/vfb/meson.build xserver-xlibre-xserver-25.0.0.22/hw/vfb/meson.build
--- xserver-xlibre-xserver-25.0.0.22.orig/hw/vfb/meson.build	2026-04-21 20:22:22.000000000 +0200
+++ xserver-xlibre-xserver-25.0.0.22/hw/vfb/meson.build	2026-05-20 07:57:01.034177394 +0200
@@ -5,20 +5,32 @@
     '../../mi/miinitext.h',
 ]
 
+vfb_link_with = [
+    libxserver_main,
+    libxserver_fb,
+    libxserver,
+    libxserver_xkb_stubs,
+    libxserver_xi_stubs,
+]
+
+vfb_c_args = []
+
+if build_glx and dri_dep.found()
+    vfb_link_with += [
+        libxserver_glx,
+        libglxvnd,
+    ]
+else
+    vfb_c_args += '-DVFB_NO_GLX'
+endif
+
 xvfb_server = executable(
     'Xvfb',
     srcs,
     include_directories: inc,
     dependencies: common_dep,
-    link_with: [
-        libxserver_main,
-        libxserver_fb,
-        libxserver,
-        libxserver_xkb_stubs,
-        libxserver_xi_stubs,
-        libxserver_glx,
-        libglxvnd,
-    ],
+    link_with: vfb_link_with,
+    c_args: vfb_c_args,
     install: true,
 )
 
